.github/workflows/template.python-wheels-ci.yml

```diff
@@ -0,0 +1,67 @@
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: Python Wheels CI
+
+on:
+  workflow_call:
+
+permissions: read-all
+
+jobs:
+  linux:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout the repository
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+          persist-credentials: false
+      - name: Build wheels
+        run: |
+          cd flink-python
+          bash dev/build-wheels.sh
+      - name: Tar artifact
+        run: tar -cvf python-wheel.tar flink-python/dist
+      - name: Upload artifact
+        uses: actions/upload-artifact@v4
+        with:
+          name: wheel_${{ runner.os }}_${{ github.sha }}_${{ github.run_number }}
+          path: python-wheel.tar
+  macos:
+    runs-on: macos-latest
+    steps:
+      - name: Checkout the repository
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+          persist-credentials: false
+      - name: Install python
```

Comment:
I'm wondering whether we should also add the explicit python installation to Linux. To have both jobs use the same Python version. Have you checked that this is working on the GHA runners (i.e. is Python installed on Linux but not MacOS)?

Reply:
Yes good idea, I took the same steps from Azure pipelines. But I agree having same Python version on both Linux/MacOs jobs is good idea. I will change workflow.

I didn't check which version is installed in the Linux VM